staff/views/index.py

```diff
@@ -0,0 +1,80 @@
+import functools
+import itertools
+import operator
+
+from django.db.models import Q
+from django.template.response import TemplateResponse
+from django.utils.decorators import method_decorator
+from django.views.generic import View
+
+from jobserver.authorization import CoreDeveloper
+from jobserver.authorization.decorators import require_role
+from jobserver.models import Backend, User
+
+
+# configure searchable models here, each must have get_staff_url defined
+configured_searches = [
+    {"model": Backend, "fields": ["name", "slug"], "order_by": "name"},
+    {
+        "model": User,
+        "fields": ["first_name", "last_name", "username"],
+        "order_by": "username",
+    },
+]
+
+
+def get_results(q):
+    """
+    Turn a search term into queries for each configured model
+
+    This takes the given search term and for each model configured to be
+    searchable does:
+
+        1. creates a Q(field=search_term) instance
+        2. combines them with a bitwise OR, typically expressed with a | (pipe)
+        3. creates a QuerySet for the model
+        4. adds that QuerySet to a list for later
+
+    The list of QuerySets is consumed, turning it in a flat list (via a
+    generator) of Model isntances across all searchable models which returned
+    results.
+
+    """
+    if not q:
+        return []
+
+    queries = []
+    for target in configured_searches:
+        qwargs = functools.reduce(
+            operator.or_, (Q(**{f"{f}__icontains": q}) for f in target["fields"])
```

Comment:
not even a nit: You can also do `Q.__or__`, to avoid an import!

Reply:
TIL 👍